pkg/client/record/event_test.go

```diff
@@ -165,6 +177,96 @@ func TestEventf(t *testing.T) {
 	}
 }
 
+func TestDuplicateEventf(t *testing.T) {
+	testPod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			SelfLink:  "/api/v1beta1/pods/foo",
+			Name:      "foo",
+			Namespace: "baz",
+			UID:       "bar",
+		},
+	}
+	testRef, err := api.GetPartialReference(testPod, "desiredState.manifest.containers[2]")
+	if err != nil {
+		t.Fatal(err)
+	}
+	item := struct {
+		obj        runtime.Object
+		reason     string
+		messageFmt string
+		elements   []interface{}
+		expect     *api.Event
+		expectLog  string
+	}{
+		obj:        testRef,
+		reason:     "Started",
+		messageFmt: "some verbose message: %v",
+		elements:   []interface{}{1},
+		expect: &api.Event{
```

Comment:
Can you test more variations of expect and the output?  Some negative tests proving that similar but different events are not combined would be good.